core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/ChunkBuffer.java

```diff
@@ -88,6 +86,30 @@ public synchronized Iterable<ByteBuffer> split(ByteBuffer inputByteBuffer) {
         return byteBuffers;
     }
 
+    /**
+     * Split the input {@link ByteBuffer} into multiple smaller {@link ByteBuffer}s, each of which contains {@link #chunkSize}
+     * worth of bytes, when the total length is unknown.
+     */
+    public synchronized Iterable<ByteBuffer> splitWithUnknownLength(ByteBuffer buffer) {
+        List<ByteBuffer> bufferedList = new ArrayList<>();
+        while (buffer.hasRemaining()) {
+            int bytesToCopy = Math.min(buffer.remaining(), currentBuffer.remaining());
+            byte[] bytes = new byte[bytesToCopy];
+            buffer.get(bytes);
+            currentBuffer.put(bytes);
+
+            if (!currentBuffer.hasRemaining() || !buffer.hasRemaining()) {
+                currentBuffer.flip();
+                ByteBuffer bufferToSend = ByteBuffer.allocate(currentBuffer.limit());
```

Comment:
How can we guarantee all ByteBuffers (except the last one) we send have the same chunk size?